tensorflow_addons/seq2seq/beam_search_decoder.py

```diff
@@ -717,6 +717,13 @@ def initialize(self, embedding, start_tokens, end_token, initial_state):
         self._start_tokens = tf.tile(
             tf.expand_dims(self._start_tokens, 1), [1, self._beam_width])
         self._start_inputs = self._embedding_fn(self._start_tokens)
+        
+        # We need to prevent the RNN cell from caching the dropout masks
+        # since they will have different shapes for a BasicDecoder
+        # and a BeamSearchDecoder sharing the same cell
+        if self._cell._cell.dropout > 0.0:
```

Comment:
note that the _cell might not have dropout if it is a customize cell. Please do some type or method existence check.